### PR TITLE
Add a snapshot json string example in examples dir

### PR DESCRIPTION
**Pull request by jinqi7**

### CODE REVIEW

File: examples/snapshot_json_string_example [hunk 1]
```diff
@@ -0,0 +1 @@
+{"application":"application-sample","components":[{"name":"component-sample","containerImage":"quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9","source":{"git":{"url":"[git@github.com](mailto:git@github.com):redhat-appstudio/integration-service.git\n","revision":"sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c"}}
```
Comment (dirgim) [inline examples/snapshot_json_string_example]: Do we need this string version file or can we just use the .json file to show the example?

File: examples/snapshot.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"application":"application-sample","components":[{"name":"component-sample","containerImage":"quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9","source":{"git":{"url":"[git@github.com](mailto:git@github.com):redhat-appstudio/integration-service.git\n","revision":"sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c"}}
```
Comment (dirgim) [inline examples/snapshot.json]: I think that the formatting modified the github link, I think it should look something like this:
```suggestion
{"application":"application-sample","components":[{"name":"component-sample","containerImage":"quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9","source":{"git":{"url":"git@github.com:redhat-appstudio/integration-service.git","revision":"sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c"}}
```

Comment (jinqi7) [reply examples/snapshot.json]: Updated. Thanks

File: examples/snapshot_json_string_example (hunk 1 above)
Comment (jinqi7) [reply examples/snapshot_json_string_example]: The parameter [SNAPSHOT](https://github.com/redhat-appstudio/integration-service/blob/main/tekton/integration_pipeline.go#L152C2-L152C2) is to contain the Snapshot json string.

Comment (dirgim) [reply examples/snapshot_json_string_example]: Ok, let's keep it that way then.

File: examples/snapshot.json (hunk 2 above)
Comment (dirgim) [reply examples/snapshot.json]: Not seeing the updated changes, can you check if it looks ok on your side?

**Comment by hongweiliu17** [comment]
@jinqi7 Sorry, I might miss something, where these files will be used? Thanks!

**Comment by jinqi7** [comment]
> @jinqi7 Sorry, I might miss something, where these files will be used? Thanks!

@hongweiliu17 , in https://issues.redhat.com/browse/STONEINTG-489 task.

File: examples/snapshot_json_string_example [hunk 3]
```diff
@@ -0,0 +1 @@
+{"application":"application-sample","components":[{"name":"component-sample","containerImage":"quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9","source":{"git":{"url":"git@github.com:redhat-appstudio/integration-service.git\n","revision":"sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c"}}}],"artifacts":{}}
```
Comment (hongweiliu17) [inline examples/snapshot_json_string_example]: Does that `git@github.com:redhat-appstudio/integration-service.git\n` need this `\n`?

Comment (jinqi7) [reply examples/snapshot_json_string_example]: It's created automatically by our code. This is copied from our cluster. So, I think "yes". It's needed.

Comment (dirgim) [reply examples/snapshot_json_string_example]: Hmmm, the `\n` is not supposed to be there, we'll probably need to investigate why it's showing up in the build pipeline results...

File: examples/pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - description: 'Snapshot of the application'
+      name: SNAPSHOT
+      default: '{"components": [{"name":"test-app", "containerImage": "quay.io/example/repo:latest"}]}'
+      type: string
+    - description: 'Namespace where the application is running'
+      name: NAMESPACE
+      default: "default"
+      type: string
+    - name: test-task
+      taskRef:
```
Comment (jsztuka) [inline examples/pipeline.yaml:18]: How does this work? Setting taskRef withou actual reference to task? I guess you ment to use taskSpec here?

File: examples/task.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+kind: Task
+metadata:
+  name: test-app
+spec:
+  description: >-
+    Expects an Snapshot deployed.
+  params:
+    - name: SNAPSHOT
+      description: The json string of the Snapshot under test 
+    - default: 'default'
```
Comment (jsztuka) [inline examples/task.yaml:11]: I'm not sure whether the `name` should be first there

File: examples/snapshot_json_string_example (hunk 3 above)
Comment (jinqi7) [reply examples/snapshot_json_string_example]: OK.  I removed it.

File: examples/pipeline.yaml (hunk 4 above)
Comment (jinqi7) [reply examples/pipeline.yaml:18]: Thanks. The "test-task" is in task.yaml now.

File: examples/snapshot_json_string_example (hunk 3 above)
Comment (jinqi7) [reply examples/snapshot_json_string_example:1]: In the code, "snapshotString, _ := json.Marshal(snapshot.Spec)" is used to create the json string. And the snapshot.Spec is like this -
spec:
  application: application-sample
  artifacts: {}
  components:
    - containerImage: >-
        quay.io/redhat-appstudio/sample-image@sha256:34fb1c72bc976937d0a59849160746cec12a8c925f31e74dc9f3b922f74da0e9
      name: component-sample
      source:
        git:
          revision: >-
            sha256:3703843e80f39188c060a6ed0f524519a682be1e7141fee29b878b1973e9783c
          url: |
            git@github.com:redhat-appstudio/integration-service.git
status:  -->  ignore this line, just show the end of the spec

Comment (dirgim) [reply examples/snapshot_json_string_example]: Thanks!

**Review by dirgim — APPROVED** [review]
LGTM

File: examples/task.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: test-app
```
Comment (MartinBasti) [inline examples/task.yaml:4]: test-task ?

File: examples/snapshot_json_string_example (hunk 3 above)
Comment (hongweiliu17) [reply examples/snapshot_json_string_example]: @jinqi7  `\n` appears again.

**Review by hongweiliu17 — APPROVED** [review]
Only one comment for `\n`, others look good to me.

File: examples/task.yaml (hunk 6 above)
Comment (jinqi7) [reply examples/task.yaml:4]: Yes, it's the name of the task. Any issue?

Comment (MartinBasti) [reply examples/task.yaml:4]: I meant to rename task, to `test-task`, so it aligned with `test-pipeline` name

**Pull request merged**